vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -899,14 +904,59 @@ class OnboardingViewModel @AssistedInject constructor(
             val activeSession = activeSessionHolder.getActiveSession()
             try {
                 activeSession.setDisplayName(activeSession.myUserId, displayName)
-                setState { copy(asyncDisplayName = Success(Unit)) }
+                setState {
+                    copy(
+                            asyncDisplayName = Success(Unit),
```

Comment:
Can't it be `asyncDisplayName = Success(displayName)` instead of storing the value in `personalizationState`. The remark is also applicable to `selectedPictureUri` I guess.

Reply:
this is a good point, at the moment those asyncs are only being used for showing/hiding the loading view, in `Login2` they're replaced with a single `isLoading: Boolean`, I was thinking of doing the same but wanted to do all the asyncs at the same time  

WDYT?

Reply:
That sounds like a plan! OK for me. What I want to avoid is having multiple sources of truth.

Reply:
will raise a PR for it :100: